tensorflow/python/estimator/canned/head.py

```diff
@@ -263,7 +263,7 @@ def _check_dense_labels_match_logits_and_reshape(
         if (dim1 is not None) and (dim1 != expected_labels_dimension):
           raise ValueError(
               'Mismatched label shape. '
-              'Classifier configured with n_classes=%s.  Received %s. '
+              'Expected labels dimension=%s.  Received %s. '
```

Comment:
please also mention that this shape is expected due to n_classes

Reply:
For one-hot lables, 
```
[[0, 0, 1],
  [0, 1, 0],
  [1, 0, 0]]
```
`expected_labels_dimension` keeps match with `n_classes`.

However, for dense labels,
```
[[0],
 [1],
 [2]]
```
`n_classes` seems irrelevant for `expected_labels_dimension`.

Reply:
ah you're right i was confused. could you please mention n_classes as an useful information in error message?

Reply:
I'm afraid not, because `_Head` class contains only logits, features, and labels, without `n_classes` information. However, I rewrite the suggestion to guide user how to fix the problem correctly.